src/canvas.class.js

```diff
@@ -1491,21 +1491,9 @@
         this.fire('selection:cleared', { target: obj });
         obj.fire('deselected');
       }
-      // TODO: need a way to cleanly check if obj is one of _hoveredTargetN and de-ref it
-      // if (obj === this._hoveredTarget){
-      //   this._hoveredTarget = null;
-      // }
-      // is there a more sane way than looping through ALL properties of *this* ?
-      // i wanted to put them into a ._hoveredTargets[Array]
-      // but, that complicates the logic of fireSyntheticInOutEvents
-      var keys = Object.keys(this);
-      for (var i = 0; i < keys.length; i++){
-        var key = keys[i];
-        if (key.indexOf('_hoveredTarget') > -1){
-          if (obj === this[key]){
-            this[key] = null;
-          }
-        }
+      if (obj === this._hoveredTarget){
+        this._hoveredTarget = null;
+        this._hoveredTargets = [];
```

Comment:
seems like the correct thing to do here is clear out _hoveredTargets as well, yes? should we be firing mouseout on these here?

Reply:
well that is not important for now. Seems right to delete them since _hoveredTargets are just part of the main target, if the main target goes, also _hoveredTargets goes.